pkg/oauthserver/oauth/external/gitlab/gitlab_oidc.go

```diff
@@ -61,6 +64,7 @@ func NewOIDCProvider(providerName, URL, clientID, clientSecret string, transport
 		PreferredUsernameClaims: []string{gitlabPreferredUsernameClaim},
 		EmailClaims:             []string{gitlabEmailClaim},
 		NameClaims:              []string{gitlabDisplayNameClaim},
+		GroupsClaims:            []string{gitlabGroupsClaim},
```

Comment:
is it safe to pull groups from GitLab automatically?

Reply:
are they guaranteed to be a list ?

Reply:
For GitLab yes, but in general OIDC does not require that.  The OIDC code handles both strings and string arrays.

Reply:
then I would make common code for OIDC and reuse it here, GitLab may always change it later and be OIDC spec compatible, we should not be caught failing if that happens

Reply:
The GitLab code reuses the OIDC code just with a specific configuration.  Thus it already handles both cases.